src/deploy/functions/deploy.ts

```diff
@@ -47,18 +63,11 @@ export async function deploy(
   options: Options,
   payload: args.Payload
 ): Promise<void> {
-  if (!context.config) {
-    return;
-  }
-
-  if (!context.functionsSourceV1 && !context.functionsSourceV2) {
-    return;
-  }
-
+  assertPreconditions(context, options, payload);
```

Comment:
I don't think this is correct. The previous code helped make sure that we silently short-circuit the functions deploy code if someone makes a deploy but the functions directory doesn't exist. At least verify that this works for a new project w/o inlitializing functions and one where the functions directory exists but the samples are still commented out & therefore no functions are to be deployed.

Reply:
Oh interesting. I thought returning early was a mistake - I'll double check the behavior and bring back the old behavior.

Reply:
Deploy unexpectedly failed on a project w/ functions directly with 0 function. Reverted back to the old logic. Thanks for the catch!